src/App.jsx

```diff
@@ -1,17 +1,40 @@
+import { useState } from 'react';
 import './App.css';
+import messages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [chatMessages, setChatMessages] = useState(messages);
+
+  const handleLikeToggle = (id) => {
+    setChatMessages((prevMessages) =>
+      prevMessages.map((message) =>
+        message.id === id
+          ? { ...message, liked: !message.liked }
+          : message
+      )
+    );
+  };
```

Comment:
Nice work on this function! By passing this down to your individual chat messages you are now able to have a single source of truth!